File: ProjetAttractionAngular/src/app/model/boutique.ts
import { Construction } from './construction';

export class Boutique extends Construction {

  constructor(
    private _revenuJourPersonne : number,
	  nom : string,
	  taille: number,
	  prixAcquisition: number,
	  prixFonctionnement: number,
	  nbAmelioration: number,
	  tauxIncident: number,
	  affluence: number,
	  id?: number
  )
  {super(nom,taille,prixAcquisition,prixFonctionnement,
    nbAmelioration,tauxIncident,affluence,id);}


	public get revenuJourPersonne() :number {
		return this._revenuJourPersonne;
	}

	public set revenuJourPersonne(revenuJourPersonne: number) {
		this._revenuJourPersonne = revenuJourPersonne;
	}
}

File: ParcAttractionBoot/src/main/java/ParcAttractionBoot/model/Boutique.java
package ParcAttractionBoot.model;

import javax.persistence.Column;
import javax.persistence.Entity;

import com.fasterxml.jackson.annotation.JsonView;

@Entity
public class Boutique extends Construction {

	@JsonView(JsonViews.Common.class)
	@Column(name="revenu_jour_personne")
	private double revenuJourPersonne;

	public Boutique(Integer id, String nom, double prixAcquisition, double prixFonctionnement,
			int nbAmelioration, double tauxIncident, double taille, int affluence,double revenuJourPersonne) {
		super(id,nom, prixAcquisition, prixFonctionnement,nbAmelioration, tauxIncident, taille, affluence);
		this.revenuJourPersonne = revenuJourPersonne;
	}
	
	
	public Boutique(String nom, double prixAcquisition, double prixFonctionnement,
			int nbAmelioration, double tauxIncident, double taille, int affluence,double revenuJourPersonne) {
		super(nom, prixAcquisition, prixFonctionnement,nbAmelioration, tauxIncident, taille, affluence);
		this.revenuJourPersonne = revenuJourPersonne;
	}
	
	
	public Boutique() {
		super();
	}
	

	public double getrevenuJourPersonne() {
		return revenuJourPersonne;
	}

	public void setrevenuJourPersonne(double revenuJourPersonne) {
		this.revenuJourPersonne = revenuJourPersonne;
	}	
	
	@Override
	public String toString() {
		return "Le magasin " + nom + " (id--> " + id+")\n"+"\t|Taille: " + taille + " m²"+"\t|Prix d'acquisition:" + prixAcquisition
				+ "€"+"\t|Prix quotidient: " + prixFonctionnement + "€"
				+ " sur " + nbAmelioration + "\t|Taux d'incident:" + tauxIncident + "%\t|Affluence journalière max: "
				+ affluence +" personnes"+"\n\t|Revenu journalier par personne: " + revenuJourPersonne + "€\n";
		
	}
	  
	  
}
